### PR TITLE
OMERO.matlab: always cast the group to a java.lang.String

### PR DESCRIPTION
**Pull request by sbesson**
This PR fixes a bug encountered while working on MultiMOT with the new multi-group functionality included in OMERO.matlab 5.1.4.

When retrieving objects in a different group than the session group, the method call resulted in exceptions of type `java.lang.Character cannot be cast to java.lang.String`. This exception was not thrown when passing `-1` as the group.
This PR reviews all usages of the group context in OMERO.matlab and unconditionally casts the `groupId` as a `java.lang.String`. The training examples are updated to also read object in a specified group. /cc @emilroz

### CODE REVIEW

File: examples/Training/matlab/ReadData.m [hunk 1]
```diff
@@ -245,6 +276,26 @@
...
+    for i = 1 : numel(screens),
+        print_object(screens(i));
+    end
+    fprintf(1, '\n');
+
+    % Retrieve all the screens owned by the session user in the
+    % context of the specified group
+    disp('Retrieving screens owned by the session user in a different group')
+    screens = getScreens(session, 'group', group2);
+    fprintf(1, '  Found %g plates\n', numel(screens));
```
Comment (mtbc) [inline examples/Training/matlab/ReadData.m]: really a plate count?

**Comment by sbesson** [comment]
While https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-merge-training didn't fail with this commit, I realized I passed the group name instead of the group identifiers. Will push another commit.

**Comment by sbesson** [comment]
See https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-training/131/console

**Comment by ximenesuk** [comment]
Changes look sensible and locally the `ReadData.m` script ran fine once I had set the `ice.config` correctly!

**Pull request merged**